Journeys/Email integration/Promise date change communication.md

```diff
@@ -0,0 +1,86 @@
+# Integration guidelines for OMS and Marketing Automation Platforms: Promise Date Change Communication 
+
+
+## Overview
+This document outlines the steps to ensure accurate promise date change communication from the Order Management System (OMS) to the Marketing Automation Platform. This integration enhances the customer experience and reduces manual efforts.
+
+### Step 1: Identify Promise Date Changes
+
+The BulkSendPromiseDateChange service in OMS identifies all the sales order items whose promise dates have changed in the last 30 minutes. The frequency of this service is configurable as per requirement, but it is recommended to set it to 30 minutes.
+
+#### Service Details:
+```
+<service name="bulkSendPromiseDateChangedNotification" engine="java"
+             location="co.hotwax.customerservice.order.CsrOrderServices" invoke="bulkSendPromiseDateChangedNotification" auth="true" transaction-timeout="3600">
+        <description>Bulk service to send promised date changed notification. Default send promised date changed on the last day.</description>
+        <attribute name="fromDate" type="Timestamp" mode="IN" optional="true"/>
+        <attribute name="toDate" type="Timestamp" mode="IN" optional="true"/>
+</service>
+```
+| Parameter | Description | Required |
+|-----------|-------------|----------|
+| `fromDate` | The start time for the service to read the change | No |
+| `toDate` | The end time for the service to read the change| No |
+
+### Step 2: Prepare for Email Notification
+
+The BulkSendPromiseDateChange service picks up all the updated sales order items. This service processes the data and prepares it for the email notification by bringing all the sales order items whose promise dates have changed between `fromDate` and `toDate`.
+
+### Step 3: Call `sendDeliveryDateChangedNotification` service
+After bringing all the updated items in response, the BulkSendPromiseDateChange service calls the `sendDeliveryDateChangedNotification` service for each item.
+
+#### Service Details:
+```
+<service name="sendDeliveryDateChangedNotification" engine="java" require-new-transaction="true" max-retry="3"
+             location="co.hotwax.warehouse.shipping.WarehouseServices" invoke="sendDeliveryDateChangedNotification">
+        <description>Send an order item Delivery Date Changed notification</description>
+        <implements service="orderNotificationInterface"/>
+        <attribute name="orderId" type="String" mode="IN" optional="true"/>
+        <attribute name="orderItemSeqId" type="String" mode="IN" optional="true"/>
+</service>
+```
+| Parameter | Description | Required |
+|-----------|-------------|----------|
+| `orderId` | The ID of the order in Shopify | No |
+| `orderItemSeqId` | The ID of the sequence of an item in the order | No |
```

Comment:
how can both of these be optional? one of these has to be required, and I think both are.

Reply:
Confirmed again with Nameet sir, orderID is required

Reply:
make a ticket for this in the OMS git